buildcheck/views/reviewer_validation.py

```diff
@@ -3,51 +3,72 @@
 from buildcheck.components.navbar import navbar
 from buildcheck.components.footer import footer
 from buildcheck.components.status_tag import status_tag
+from buildcheck.state.user_state import UserState
 from buildcheck.backend.supabase_client import supabase_client
 
+class AIValidationState(UserState):
+    violations: list[str] = []
+    guidelines: list[dict] = []
+    case_data: list[dict] = []
+    case_result: str = "No cases assigned"  # Default case result status
+    case_id: str = ""
+    listOfCases: list[str] = []
+    comments: dict = {}
 
-class GuidelineState(rx.State):
-    guideline: list[str]  # Explicitly type as list of list of strings
-    # I know that we should'nt use this table, but I'm only using it to get guidelines examples
-    # When we finalize the DB schema, I will replace this with the correct table
-    response = supabase_client.table("guidelines").select("*").execute()
-    guideline = [
-        [item.get("code"), item.get("title"), item.get("description"),item.get("category")]
-        for item in response.data
-    ]
-class SelectState(rx.State):
-    value: str = "444"
+    @rx.event
+    def load_data(self):
+        # Loads the case data for the current user from the database
+        try:
+            user_id_int = int(self.user_id)
+            response1 = supabase_client.table("cases").select("*").eq("reviewer_id", user_id_int).execute()
+            self.case_id = str(response1.data[0]["id"])
+            self.case_data = response1.data
```

Comment:
This code throws an error when the reviewer has no cases assigned. I logged in as `laila` who has no cases.
Also, this code assumes there is only 1 case assigned to the reviewer. Are we making that assumption? If so we need to document that and put some kind of `unique` constraint. (also the query should be `limit(1).single()` then)